mistnet.R

```diff
@@ -0,0 +1,184 @@
+library(tidyverse)
+library(mistnet)
+library(progress)
+devtools::load_all()
+
+if (!file.exists("observer_model.rds")) {
+  fit_observer_model()
+}
+
+bbs = get_bbs_data() %>% 
+  mutate(species_id = paste0("sp", species_id), presence = abundance > 0) %>% 
+  select(-lat, -long, -abundance) %>% 
+  spread(key = species_id, value = presence, fill = 0)
+
+# Mistnet settings based on the second-best model in Appendix D of 
+# the mistnet paper
+n.minibatch = 23L
+latent_dim = 6L
+n.importance.samples = 24L
+N1 = 100L
+N2 = 12L
+
+# Other mistnet settings
+n_opt_iterations = 1E4
+iteration_size = 10
+n_prediction_samples = 20
+
+fit_mistnet = function(iter,
+                       use_obs_model){
+  
+  settings = yaml::yaml.load_file("settings.yaml")
+  
+  # Collect data for this iteration, then discard the rest to save
+  # memory
+  obs_model = readRDS("observer_model.rds")
+  data = filter(obs_model$data, iteration == iter) %>% 
+    left_join(bbs, c("site_id", "year"))
+  rm(obs_model)
+  gc()
```

Comment:
You don't need to do this here since this is working, but there are also a couple of other ways that never read the full file into memory, e.g., https://stackoverflow.com/questions/23197243/how-do-i-read-only-lines-that-fulfil-a-condition-from-a-csv-into-r. The `sqldf` approach seems particularly light weight and is apparently pretty faster than the alternatives.